codegenerator/cli/templates/static/codegen/src/eventFetching/ChainManager.res

```diff
@@ -296,10 +314,15 @@ let popBatchItem = (
 
 let getFetchStateWithData = (self: t, ~shouldDeepCopy=false): ChainMap.t<fetchStateWithData> => {
   self.chainFetchers->ChainMap.map(cf => {
-    partitionedFetchState: shouldDeepCopy
-      ? cf.fetchState->PartitionedFetchState.copy
-      : cf.fetchState,
-    heighestBlockBelowThreshold: cf.currentBlockHeight - cf.chainConfig.confirmedBlockThreshold,
+    {
+      partitionedFetchState: shouldDeepCopy
+        ? cf.fetchState->PartitionedFetchState.copy
+        : cf.fetchState,
+      heighestBlockBelowThreshold: Pervasives.max(
+        cf.currentBlockHeight - cf.chainConfig.confirmedBlockThreshold,
+        0,
+      ),
```

Comment:
This is just an extra guard, since with unordered multichain it could call this function while a chains "currentBlockHeight" was still at it's initial value of 0.